In [ ]:
"""
Food Safety

Food safety is one of the big challenges for companies that sell packaged food. They always receive complaints from dissatisfied customers. However, an increase in the level of the complaints might be an issue for the reputation of the company. When the food safety-related issues need to be taken very seriously. For any company, it is important to handle such issues proactively to save the reputation of the company.

A company sells prepackaged dinners. Normally it receives one or two calls each month from customers stating that they fell sick after consuming the food. Usually, they fell sick not because of food but because of some other environmental infections. If the call volume increases from 1.5 calls per month to 6 calls per month, there is a serious problem. It means that during a normal period the company receives 1.5 calls per month and during the problem period the company receives 6 calls per month. Recently, the company starts receiving 3 calls per month. The

senior management wants to know whether there are chances of a problem.

As a data scientist, you have been assigned to analyze the problems that the food company is facing.

    1. What type of random variable would be suited to model the call volume during the normal period and the problem period and why?
    2. Whether it would be surprising to get more than 3 calls a month during the problem period?
    3. Whether it would be surprising to get more than 3 calls a month during the normal period?
    4. Prepare your analysis and explain why or why not the management should take corrective actions.

"""

# 1. What type of random variable would be suited to model the call volume during the normal period and the problem period and why?

""" 



Poisson random variable are well suited for this case study. 
we have ratio 

no. of calls in normal period = 1.5

no. of calls in problem period = 6


"""

# 2. Whether it would be surprising to get more than 3 calls a month during the problem period?

"""

No, it wouldnt be suprising to get more than 3 calls in the problem period. Its becouse during that period peoples most often get sick due to the change in the environment(Environmental Infection). Therefore we should not take the calls seriously during the problem period.....

"""

# 3. Whether it would be surprising to get more than 3 calls a month during the normal period?

"""

Yes, It would be a little suprising to get more than 3 calls during the normal period becouse during the problem period its more often due to the environmental infection but to get an increase in the no. of calls in the normal period is a Serious ISSUE!!!!.


So, the Company should check the quality of the food to be sure that everything is fine from their side but if there are some problem with the quality then the company should take some strict steps so that this thing wont occur again inorder to maintain their status in the Market!!!. 


"""

# 4. Prepare your analysis and explain why or why not the management should take corrective actions.

In [4]:
#p(more than 3 calls) in normal Period

from scipy.stats import poisson
mu=1.5
print(1-poisson.cdf(3,mu))
# It is not normal to get more than 3 calls in Normal period!!!

0.06564245437845007


In [5]:
#P(more than 3 calls) in problem period
from scipy.stats import poisson
mu=6
print(1-poisson.cdf(3,mu))
# It is normal to get more than 3 calls in problem period!!1

0.8487961172233521


"""

It is rare to get many calls during a month. It happens only 6.6% in the month 
and during a problem time more than 3 calls a month is common 84.87%


"""